### ISSUE TITLE
Self-hosted bot stops posting feed updates until restarted

### ISSUE THREAD

**Issue by rzumer** [synzen/MonitoRSS#196]
**Describe the bug**
I am self-hosting the bot using my own MongoDB database on one machine and the bot on another (ran through the quick start method).

After a while the bot stops posting any feed update despite fetching them correctly according to the log, and responding to commands. It needs to be restarted to react to feed updates again. It seems to stop working after a few hours (less than a day), but I don't know if it is a constant amount of time.

**To Reproduce**
`nohup node start.js > monitorss.log &`

```
const MonitoRSS = require('monitorss')

const config = {
  bot: {
    token: '[redacted]',
    prefix: 'rss.',
    ownerIDs: ['[redacted]'],
    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0'
  },
  database: {
    uri: 'mongodb://[redacted]:27017/rss'
  },
  feeds: {
    refreshRateMinutes: 5
  }
}

const settings = {
  setPresence: true,
  config
}

const client = new MonitoRSS.ClientManager(settings)
client.start()
```

**Expected behavior**
Bot posts feed updates after up to 5 minutes.

**Actual behavior**
Bot stops posting updates despite the log showing "Finished feed retrieval cycle" messages periodically with no errors, and the bot reacting to commands.

**Branch (please specify one):**
master

**Additional context**
I need to self-host to get around some websites' range IP blocks.

My database is on a remote machine, but it's an always-on dedicated server and there are no connection error messages in the log.

**Comment by rzumer** [comment]
I found that the bot suddenly loses track of its feeds. It continues queuing feed retrieval cycles but prints "No feeds to retrieve" when it is in this error state. There is no error between successful cycles and this.

**Comment by synzen** [comment]
Thanks for the details. I suspect that the feed may be erroneously returning a 304 Not Modified response in that case. You can get in-depth details about every fetch and comparison for a particular feed if you use the `debug` [owner command](https://docs.monitorss.xyz/configuration/discord-commands#bot-owner-commands). I neglected to document it before, but it's in now. 

To get the feed ID, use `rss.backup` and get the `_id` field of a feed.

**Comment by rzumer** [comment]
This happens for multiple feeds (all those that I registered with the bot, over 4-5 websites), and only after some amount of time (i.e. feed updates work for a while, until they don't). Does that sound like usual behaviour for the issue you are referring to? It sounds different to me since it doesn't seem to be feed-specific. Either way, I will try the debug command and update with any further details if I notice something.

**Comment by rzumer** [comment]
This is probably a different issue, but my feeds just all disappeared for the second time with no indication of any error or action in the log.
```
[2020-12-27 17:02:58.628 -0500] INFO    : [default] Finished feed retrieval cycle (10). Cycle Time: 5.20s
[2020-12-27 17:07:54.304 -0500] INFO    : [default] Finished feed retrieval cycle. No feeds to retrieve
```

As for the initial issue I described, since adding a feed to the debug list I have not had it occur again.

**Comment by synzen** [comment]
Debugged feeds don't modify/change anything, it just enables suppressed console logs

I can't speak much about feeds disappearing - it grabs feeds *directly* from your database/local file system, so if things are being deleted there, then the bot will reflect just that

**Comment by rzumer** [comment]
I had yet another issue come up today where the bot stops responding completely even though it keeps fetching feeds and remains marked as online.

**Comment by rzumer** [comment]
I had my feeds disappear again. When I restored them from json the bot returned `Cannot read property 'refreshRateMinutes' of undefined` on `list` command until restarted. Once again between retrieval cycles feeds completely disappeared without any message in the log.

**Comment by rzumer** [comment]
Today the bot did not post a feed update well after multiple cycles even though the log indicates that it fetched feed properly and responds to commands.

In summary it seems totally random when it starts failing, the logs may or may not indicate anything strange, and the bot may or may not respond to commands. Maybe the logs are not detailed enough to catch all issues, but it seems totally unreliable unless I restart it regularly, maybe every few hours.

**Comment by synzen** [comment]
Apologies for the late reply. I suggest you look at database logs from around the time they disappear.

And also to leave it off, and wait to see if the database data persists.

**Comment by rzumer** [comment]
It turned out my db disabled access control out of the box (maybe I had an old config) and it got repeatedly hit by some automated script clearing it and demanding a ransom, which got buried in the logs. So that explains my issue with disappearing feeds at least.

**Comment by synzen** [comment]
Is this issue resolved? If the original issue still exists with you not receiving articles, please be sure to follow the instructions I laid out in https://github.com/synzen/MonitoRSS/issues/196#issuecomment-750597201

**Comment by rzumer** [comment]
I set it up to restart every hour so I can't say for sure but I haven't had any issues since I secured my database, anyway.
